extensions/sql-database-projects/src/models/options/deployOptionsModel.ts

```diff
@@ -8,250 +8,64 @@ import * as mssql from 'mssql';
-			, this.deploymentOptions.dropConstraintsNotInSource.displayName
-			, this.deploymentOptions.dropDmlTriggersNotInSource.displayName
-			, this.deploymentOptions.dropExtendedPropertiesNotInSource.displayName
-			, this.deploymentOptions.dropIndexesNotInSource.displayName
-			, this.deploymentOptions.ignoreFileAndLogFilePath.displayName
-			, this.deploymentOptions.ignoreExtendedProperties.displayName
-			, this.deploymentOptions.ignoreDmlTriggerState.displayName
-			, this.deploymentOptions.ignoreDmlTriggerOrder.displayName
-			, this.deploymentOptions.ignoreDefaultSchema.displayName
-			, this.deploymentOptions.ignoreDdlTriggerState.displayName
-			, this.deploymentOptions.ignoreDdlTriggerOrder.displayName
-			, this.deploymentOptions.ignoreCryptographicProviderFilePath.displayName
-			, this.deploymentOptions.verifyDeployment.displayName
-			, this.deploymentOptions.ignoreComments.displayName
-			, this.deploymentOptions.ignoreColumnCollation.displayName
-			, this.deploymentOptions.ignoreAuthorizer.displayName
-			, this.deploymentOptions.ignoreAnsiNulls.displayName
-			, this.deploymentOptions.generateSmartDefaults.displayName
-			, this.deploymentOptions.dropStatisticsNotInSource.displayName
-			, this.deploymentOptions.dropRoleMembersNotInSource.displayName
-			, this.deploymentOptions.dropPermissionsNotInSource.displayName
-			, this.deploymentOptions.dropObjectsNotInSource.displayName
-			, this.deploymentOptions.ignoreColumnOrder.displayName
-			, this.deploymentOptions.ignoreTablePartitionOptions.displayName
-			, this.deploymentOptions.doNotEvaluateSqlCmdVariables.displayName
-			, this.deploymentOptions.disableParallelismForEnablingIndexes.displayName
-			, this.deploymentOptions.disableIndexesForDataPhase.displayName
-			, this.deploymentOptions.restoreSequenceCurrentValue.displayName
-			, this.deploymentOptions.rebuildIndexesOfflineForDataPhase.displayName
-			, this.deploymentOptions.preserveIdentityLastValues.displayName
-			, this.deploymentOptions.isAlwaysEncryptedParameterizationEnabled.displayName
-			, this.deploymentOptions.allowExternalLibraryPaths.displayName
-			, this.deploymentOptions.allowExternalLanguagePaths.displayName
-			, this.deploymentOptions.hashObjectNamesInLogs.displayName
-			, this.deploymentOptions.doNotDropWorkloadClassifiers.displayName
-			, this.deploymentOptions.ignoreWorkloadClassifiers.displayName
-			, this.deploymentOptions.ignoreDatabaseWorkloadGroups.displayName
-			, this.deploymentOptions.doNotDropDatabaseWorkloadGroups.displayName].sort();
+	public UpdateOptionsMapTable() {
+		this.optionsMapTable = this.deploymentOptions.optionsMapTable;
 	}
 
 	/**
-	 * Gets the options checkbox check value
-	 * @returns string[][]
+	 * Gets the options checkbox values by iterating through the labels and gets the default/changed value from the optionsMapTable
+	 * Returns data as [optionName, booleanValue], where these values are sending to the options table component
```

Comment:
Big +1 to using an interface to get rid of magic indexing constants everywhere.

Reply:
I want to have an optionsValueLookup as a map, to hold optionName and its value. How can I use the above proposed interface into a map? should my interface itself a map to avoid [0] and [1]? Thanks